sorts.py

```diff
@@ -68,10 +68,33 @@ def chris_insert_sort(values):
 
 
 def peter_insert_sort(values):
-    # TODO: Test for empty input.
-    # TODO: Build sorted linked list from input using Link class.
-    # TODO: Copy sorted linked list into Python list.
-    pass
+    newlist = []
+    if len(values) <= 1:
+        return values
+    else:
+        head = Link(values[0], None)
+        current = head
+        for num in values[1:]:
+            modified = False
+            if num < head.value:
+                head = Link(num, head)
+                current = head
+            else:
+                while not modified:
+                    if current.nextval == None:
+                        current.nextval = Link(num, current.nextval)
+                        modified = True
+                    elif num < current.nextval.value:
+                        current.nextval = Link(num, current.nextval)
+                        current = head
```

Comment:
This is the third "current = head" I've seen. Why not get rid of all of them and put a single "current = head" just before what's now "while True:". That's the loop where you always want to start at the head of the list. And it's cleaner to be explicit about it at the start of each loop rather than rely on implicit leftover state from previous work.